capstone/src/main/java/com/google/sps/servlets/clubs/ClubServlet.java

```diff
@@ -29,12 +30,27 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Gson gson = new Gson();
-    String json =
-        gson.toJson(
-            PrototypeClubs.PROTOTYPE_CLUBS_MAP.get(request.getParameter(Constants.CLUB_NAME_PROP)));
-    response.setContentType("text/html;");
-    response.getWriter().println(json);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    Entity clubEntity = retrieveClub(request, datastore).asSingleEntity();
+    if (clubEntity != null) {
+      String name = clubEntity.getProperty(Constants.PROPERTY_NAME).toString();
+      ImmutableList<String> members =
+          ImmutableList.copyOf((ArrayList<String>) clubEntity.getProperty(Constants.MEMBER_PROP));
+      ImmutableList<String> officers =
+          ImmutableList.copyOf((ArrayList<String>) clubEntity.getProperty(Constants.OFFICER_PROP));
+      String description = clubEntity.getProperty(Constants.DESCRIP_PROP).toString();
+      String website = clubEntity.getProperty(Constants.WEBSITE_PROP).toString();
+      ImmutableList<String> announcements =
+          ImmutableList.copyOf((ArrayList<String>) clubEntity.getProperty(Constants.ANNOUNCE_PROP));
+      Club club = new Club(name, members, officers, description, website, announcements);
+      Gson gson = new Gson();
+      String json = gson.toJson(club);
+      response.setContentType("text/html;");
```

Comment:
Should this be "text/html" or "application/json"? I've seen it done both ways (and it doesn't seem to matter) but what's the correct way to do it?

Reply:
I'm not really sure either, but I would assume "application/json" would be the better fit here? thanks!

Reply:
I think it might have to do with performance or how the response is treated (https://stackoverflow.com/questions/14946845/performance-text-html-vs-application-json)? I think json can be accepted as regular text, which is why text/html works, but it's more efficient and safer to use application/json possibly